pennylane/workflow/get_gradient_fn.py

```diff
@@ -86,7 +96,3 @@ def _get_gradient_fn(
 
     if isinstance(diff_method, TransformDispatcher):
         return diff_method
-
-    raise qml.QuantumFunctionError(
-        f"Differentiation method {diff_method} must be a gradient transform or a string."
-    )
```

Comment:
Potentially we can move the `if isinstance(diff_method, str)` block below:

```
        raise qml.QuantumFunctionError(
            f"Differentiation method {diff_method} not recognized. Allowed "
            f"options are {tuple(get_args(SupportedDiffMethods))}."
        )
```

That way we aren't left with a hanging ending to the function.